conda/version.py

```diff
@@ -302,6 +302,10 @@ def __new__(cls, spec):
         self.spec = spec
         if isinstance(spec, tuple):
             self.match = self.all_match_ if spec[0] == 'all' else self.any_match_
+        elif spec.startswith("^") and spec.endswith("$"):
```

Comment:
@megies and I agreed that if someone is that determined to do full regex matching there's no need to handle the other operators. Just let it be standalone. After all, the other operators can be emulated in the regex itself.